crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_literal_union.rs

```diff
@@ -201,7 +209,11 @@ pub(crate) fn redundant_literal_union<'a>(checker: &Checker, union: &'a Expr) {
                             parenthesized: true,
                         })
                     } else {
-                        group[0].clone()
+                        group
+                            .first()
+                            .expect("should have at least one new_expr")
+                            .to_owned()
+                            .clone()
```

Comment:
This can still panic, just with a nicer error message. I think I was picturing something more like this:

```suggestion
                        let Some(group) = group.first() else {
                            return;
                         };
                         group //.clone() if necessary
```